python/tests-unit/test_serializer.py

```diff
@@ -70,12 +68,15 @@ def test_strings(self, source, expected):
         result = serialization.Serializer._str(source)
         assert result == expected
 
-    def test_strings_emoji_zwj(self):
-        if sys.version_info >= (3, 12):
-            raise pytest.xfail("CCT-644")
-
-        source = "👨🏼‍🚀"
+    @pytest.mark.parametrize(
+        "source,expected",
+        [
+            ("zw​space", "'zw\\u200bspace'"),
+            ("zw‌nonjoiner", "'zw\\u200cnonjoiner'"),
+            ("👨🏼‍🚀", "'👨🏼\\u200d🚀'"),
+        ],
+        ids=["zero-width space", "zero-width non-joiner", "zero-width joiner"],
```

Comment:
Note you can specify id's for the items directly when declaring them, eg
```python
            pytest.param("zw​space", "'zw\\u200bspace'", id="zero-width space"),
```
pytest supports mixing "raw values" and `pytest.param()` together.

It is not a blocking issue, and I'd like to see a followup commit that moves the data for the `test_strings_unicode` test as part of the `test_strings` test, as now the two tests are practically the same.